tests/sentry/tasks/integrations/test_link_all_repos.py

```diff
@@ -0,0 +1,200 @@
+        self.expires_at = isoformat_z(timezone.now() + timedelta(days=365))
+
+    def _add_responses(self):
+        responses.add(
+            responses.POST,
+            self.base_url + f"/app/installations/{self.installation_id}/access_tokens",
+            json={"token": self.access_token, "expires_at": self.expires_at},
+        )
+        responses.add(
+            responses.GET,
+            self.base_url + "/installation/repositories?per_page=100",
+            status=200,
+            json={
+                "total_count": 2,
+                "repositories": [
+                    {
+                        "id": 1,
+                        "full_name": "getsentry/sentry",
+                    },
+                    {
+                        "id": 2,
+                        "full_name": "getsentry/snuba",
+                    },
+                ],
+            },
+        )
+
+    @responses.activate
+    def test_link_all_repos(self, get_jwt):
+        self._add_responses()
+
+        link_all_repos(
+            integration_key=self.key,
+            integration_id=self.integration.id,
+            organization_id=self.organization.id,
+        )
+
+        repos = Repository.objects.all()
+        assert len(repos) == 2
+
+        for repo in repos:
+            assert repo.organization_id == self.organization.id
+            assert repo.provider == "integrations:github"
+
+        assert repos[0].name == "getsentry/sentry"
+        assert repos[1].name == "getsentry/snuba"
+
+    @responses.activate
+    def test_link_all_repos_api_response_keyerror(self, get_jwt):
+        responses.add(
+            responses.POST,
+            self.base_url + f"/app/installations/{self.installation_id}/access_tokens",
+            json={"token": self.access_token, "expires_at": self.expires_at},
+        )
+        responses.add(
+            responses.GET,
+            self.base_url + "/installation/repositories?per_page=100",
+            status=200,
+            json={
+                "total_count": 2,
+                "repositories": [
+                    {
+                        "full_name": "getsentry/sentry",
+                    },
+                    {
+                        "id": 2,
+                        "full_name": "getsentry/snuba",
+                    },
+                ],
+            },
+        )
+
+        link_all_repos(
+            integration_key=self.key,
+            integration_id=self.integration.id,
+            organization_id=self.organization.id,
+        )
+
+        repos = Repository.objects.all()
+        assert len(repos) == 1
+
+        assert repos[0].organization_id == self.organization.id
+        assert repos[0].provider == "integrations:github"
+
+        assert repos[0].name == "getsentry/snuba"
+
+    @patch("sentry.tasks.integrations.link_all_repos.metrics")
+    def test_link_all_repos_missing_integration(self, mock_metrics, get_jwt):
```

Comment:
nit: If you don't touch `get_jwt` anywhere, the just use `_` as an ignored argument